AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/script_canvas_tools_qt.py

```diff
@@ -0,0 +1,590 @@
+
+def remove_script_event_parameter(self):
+    remove_param = self.asset_editor_row_container.findChild(QtWidgets.QFrame, "[0]").findChild(QtWidgets.QToolButton, "")
+    remove_param.click()
+    result = helper.wait_for_condition(
+        lambda: self.asset_editor_widget.findChild(QtWidgets.QFrame, "[0]") is None, WAIT_TIME_3
+    )
+    Report.result(Tests.parameter_removed, result)
+
+def add_empty_parameter_to_script_event(self, number_of_parameters):
+    """
+    Function for adding a new blank parameter to a script event
+
+    param self: the script calling this function
+    param number_of_parameters: the number of empty parameters to add
+
+    returns none
+    """
+    helper.wait_for_condition(
+        lambda: self.asset_editor_row_container.findChild(QtWidgets.QFrame, PARAMETERS_QT) is not None, WAIT_TIME_3)
+    parameters = self.asset_editor_row_container.findChild(QtWidgets.QFrame, PARAMETERS_QT)
+    add_parameter = parameters.findChild(QtWidgets.QToolButton, "")
+
+    for _ in range(number_of_parameters):
+        add_parameter.click()
+
+def get_script_event_parameter_name_text(self):
+    """
+    function for retrieving the name field of script event parameters
+
+    param self: the script calling this function
+
+    returns a container with all the parameters' editable name fields
+    """
+    parameter_names = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, NAME_STRING)
+    name_fields = []
+    for parameter_name in parameter_names:
+        name_fields.append(parameter_name.findChild(QtWidgets.QLineEdit))
+
+    return name_fields
+
+def get_script_event_parameter_type_combobox(self):
+    """
+    function for retrieving the type field of script event parameters
+
+    param self: the script calling this function
+
+    returns a container with all the parameters' editable type combo boxes
+    """
+    parameter_types = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, "Type")
+    type_combo_boxes =[]
+    for parameter_type in parameter_types:
+        type_combo_boxes.append(parameter_type.findChild(QtWidgets.QComboBox))
+
+    return type_combo_boxes
+
+
+def located_expected_tracer_lines(self, section_tracer, lines):
+    """
+    function for parsing game mode's console output for expected test lines. requires section_tracer. duplicates lines 
+    and error lines are not handled by this function
+    
+    param self: The script calling this function
+    param section_tracer: python editor tracer object
+    param lines: list of expected lines
+    
+    
+    returns true if all the expected lines were detected in the parsed output
+    """
+    found_lines = [printInfo.message.strip() for printInfo in section_tracer.prints]
+
+    expected_lines = len(lines)
+    matching_lines = 0
+
+    for line in lines:
+        for found_line in found_lines:
+            if line == found_line:
+                matching_lines += 1
+
+    return matching_lines >= expected_lines
```

Comment:
Let's extract this to another helper at the root of EditorPythonTestTools. Maybe an existing lib or a newly created one?

Reply:
migrated this to the editor testing tools library